wiki/Client/Release_stream/Lazer/File_storage/en.md

```diff
@@ -0,0 +1,47 @@
+# File storage in osu!(lazer)
+
+By default, osu!(lazer) stores beatmaps, skins, and score replay files in the following directories:
+
+- `%appdata%/osu/files` (Windows),
+- `~/.local/share/osu/files` (Linux & macOS),
+- `Android/data/sh.ppy.osulazer/files/files` (Android).
+
+On desktop platforms, the file store can also be moved in its entirety to another location, using the `Change folder location...` button in client settings.
+
+## Storage structure
+
+osu!(stable)'s storage structure, where data was directly accessible to users as standard files, resulted in a lot of teething issues due to having to explicitly handle unwanted modifications of key files. To avoid this, osu!(lazer) employs a more stringent method of file storage. As such, there is no equivalent of the `Songs` and `Skins` folders. All files that are imported to lazer are stored under filenames that reflect their [SHA-256 hashes](https://en.wikipedia.org/wiki/SHA-2). Mappings to these files are held inside a client database.
+
+For example, a file with the SHA-256 hash of
+
+```
+1a47929b6056d34d25a95eeb2012395ceed66af6f40cc37c898a08482d6325d2
+```
+
+shall be stored under the path of
```

Comment:
```suggestion
would be stored under the path of
```

Reply:
not sure about changing to a subjunctive `would`? like, this *is* going to happen, that's what the game *does*, that's a fact and there's no need to conditionalize this i don't think. but i don't care enough to argue

Reply:
my argument is that *if* such a file exists, then it *would* be stored at so-and-so path. we're talking about a hypothetical file and not a real one

what you wrote makes sense too, but the problem with "shall" is that it's a bit too archaic for the wiki. you could use "will" instead.

Reply:
meh it's fine